enterprise/server/sociartifactstore/sociartifactstore.go

```diff
@@ -281,36 +275,44 @@ func (s *SociArtifactStore) GetArtifacts(ctx context.Context, req *socipb.GetArt
 	return &resp, nil
 }
 
-func blobKey(hash string) string {
-	return blobKeyPrefix + hash
+// Accepts a container image config hash 'h' which uniquely identifies the
+// image and generates the repb.Digest that should be used as the cache key
+// for storing the SOCI Index for that image.
+func sociIndexKey(h v1.Hash) (*repb.Digest, error) {
+	buf := bytes.NewBuffer([]byte(h.Hex + *sociIndexCacheSeed))
+	return digest.Compute(buf, repb.DigestFunction_SHA256)
 }
 
-func resourceName(digest *repb.Digest) *rspb.ResourceName {
+func resourceName(digest *repb.Digest, cacheType rspb.CacheType) *rspb.ResourceName {
 	return &rspb.ResourceName{
 		Digest:         digest,
 		Compressor:     repb.Compressor_IDENTITY,
-		CacheType:      rspb.CacheType_CAS,
+		CacheType:      cacheType,
 		DigestFunction: repb.DigestFunction_SHA256,
 	}
 }
 
-func (s *SociArtifactStore) getArtifactsFromCache(ctx context.Context, imageId string) (*socipb.GetArtifactsResponse, error) {
-	exists, err := s.blobstore.BlobExists(ctx, blobKey(imageId))
+func (s *SociArtifactStore) getArtifactsFromCache(ctx context.Context, imageConfigHash v1.Hash) (*socipb.GetArtifactsResponse, error) {
```

Comment:
b/c this is only a couple of cache lookups I think it will be pretty fast, but mostly for FYI:

Another approach here, instead of using the environment cache object directly, would be to use a client pointing to the cache and actually store your data as repb.OutputFile / repb.ActionResult via the Bytestream Read/WRite and UpdateActionResult() RPCs, similar to how Bazel would.

Then this code could call GetActionResult and the server would automatically check the existence of all of the underlying files in the directory tree for you. Probably not important for this use case but just putting it out there :)

Reply:
Hmm, good point. I added a TODO describing this a little bit, but I'll leave the implementation as is for now.